### PR TITLE
Refactoring Tasks

### PR DESCRIPTION
**Pull request by yanfali**
@zekth some overdue clean up

- catch some errors
  - routing to same keyboard while loading JSON
  - keycount getter is invalid while viewing preview
- remove store references in ControllerBottom
- initialize icons in a more consistent way
- download mime type was incorrect for JSON

### CODE REVIEW

File: src/components/ControllerTop.vue [hunk 1]
```diff
@@ -238,8 +238,7 @@ export default {
             this.updateLayout(data.layout);
             let promise = new Promise(resolve =>
               store.commit('keymap/setLoadingKeymapPromise', resolve)
```
Comment (zekth) [inline src/components/ControllerTop.vue]: Waiting for a commit wouldn't be in an action? it's more the vue design?

**Comment by yanfali** [comment]
It's used as a waiting mechanism in a different part of the application.
This sets the value only.

**Comment by yanfali** [comment]
Good catch. I'll look into it.

**Comment by yanfali** [comment]
> Key counter doesn't work at all.
> 
> image

fixed

File: src/components/ControllerTop.vue (hunk 1 above)
Comment (yanfali) [reply src/components/ControllerTop.vue]: this isn't an action, it's a signaling mechanism.

**Pull request merged**